api/layer/layer.go

```diff
@@ -351,7 +352,13 @@ func (n *layer) PutBucketACL(ctx context.Context, param *PutBucketACLParams) err
 		return err
 	}
 
-	return n.setContainerEACLTable(ctx, inf.CID, param.EACL)
+	var sessionToken *session.Token
+	boxData, err := GetBoxData(ctx)
+	if err == nil {
+		sessionToken = boxData.Gate.SessionTokenForSetEACL()
+	}
```

Comment:
Can we move this to the handlers level (retrieve session token in functions like `CreateBucketHandler`)?

Reply:
I'm not sure, because the function PutBucketACL is called in `PostObject`, `PutObjectHandler` and  `updateBucketACL` which, in turn, we call in `PutBucketACLHandler`, `PutObjectACLHandler`, `PutBucketPolicyHandler` and `PutObjectACLHandler`.  I think moving of this code will produce a lot of copy-paste. 

Also, is it nice, that we use `Client` interface function in our "inner" function `updateBucketACL`?

Reply:
> I think moving this code to the handlers level will produce a lot of copy-paste.

 We can define the separate function to this. The reason why I consider such moving is checking session token before we we start action (e.g. put object) that require SETEAСL token.

> Also, is it nice, that we use Client interface function in our "inner" function updateBucketACL?

Actually, it's not "inner" function but common part in some handlers (`PutBucketPolicyHandler`, `CompleteMultipartUploadHandler` etc.)